frontend/lib/search/location_button.dart

```diff
@@ -0,0 +1,65 @@
+import 'package:flutter/material.dart';
+import 'package:geolocator/geolocator.dart';
+
+import '../location/determine_position.dart';
+
+class LocationButton extends StatefulWidget {
+  final void Function(Position position) setCoordinates;
+
+  const LocationButton({Key key, this.setCoordinates}) : super(key: key);
+
+  @override
+  State<StatefulWidget> createState() => _LocationButtonState();
+}
+
+enum LocationRequestStatus { requesting, requestSuccessful, requestFailed }
+
+class _LocationButtonState extends State<LocationButton> {
+  LocationRequestStatus _locationStatus = LocationRequestStatus.requesting;
+
+  @override
+  Widget build(BuildContext context) {
+    var theme = Theme.of(context);
+    if (_locationStatus == LocationRequestStatus.requestSuccessful) {
+      return Container();
+    }
+    if (_locationStatus == LocationRequestStatus.requesting) {
+      return Container(
```

Comment:
This looks a bit glitchy if you already given permission. The loading spinner is shown for a split second before it disappears. If you already have the permission then it should not be shown. Maybe we can add here a delay like 200ms? Can also be moved to a separate task. What do you think?

Reply:
Yeah, I know. Strangely, if granting permission fails, the button has a nice animation back to the "extended" form.
But: If the user has already given permission and has location services enabled prior to launching the app, the user won't even see the button (as the map is the first screen and the search screen will immediately start determining the user's position). Hence, this "glitch" is most likely not too severe for the user...
I will create an issue for it though.

Reply:
I added an animation :stuck_out_tongue:  :nerd_face: